docs/cpp2/metafunctions.md

```diff
@@ -372,6 +372,67 @@ A `cpp1_rule_of_zero` type is one that has no user-written copy/move/destructor
 > This is known as "the rule of zero".
 > — Stroustrup, Sutter, et al. (C++ Core Guidelines)
 
+#### `regex`
+
+Replaces fields in the class with regular expression objects. Each field starting with `regex` is replaced with a regular expression of the same type.
```

Comment:
Is it "starting with `regex`" or is it "named `regex` or starting with `regex_`"?

Reply:
Currently it's "starting with `regex`" and it doesn't care what comes after it, it can be `regexes` if you want.

Reply:
Yes, until now I did not think about this corner case.  I will do an update that checks for `regex` and `regex_*`.

The long term goal would be to apply the metafunction to the variable definition, e.g.:
```
name_matcher: @regex = "regex here";
```